server/core_group.go

```diff
@@ -996,12 +996,16 @@ func ListGroups(ctx context.Context, logger *zap.Logger, db *sql.DB, name string
 	}
 
 	var query string
-	params := []interface{}{limit}
+	params := []interface{}{limit, omit_full}
+
 	if name == "" {
 		query = `
 SELECT id, creator_id, name, description, avatar_url, state, edge_count, lang_tag, max_count, metadata, create_time, update_time
 FROM groups
-WHERE disable_time = '1970-01-01 00:00:00 UTC'
+WHERE
+	(disable_time = '1970-01-01 00:00:00 UTC')
+AND
+	((NOT $2) OR (edge_count < max_count))
```

Comment:
I think this and subsequent `OR (edge_count < max_count)` clauses significantly change performance profile of the query. Have you done an analysis on the query? What does `ANALYSE ....` look like?

I ask, because the originally query was looking at indexed fields, and also was a key-value look up (as much as it could), and also the indexes were created in such a fashion to allow the database to fast-skip as many records as possible. However, the new query adds a range query that doesn't allow the database to skip records.